## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#52653185-79e2-430c-909b-a98854ea28c8) |
| mint-xl.sh | :x: [more...](#c67f6841-037c-4069-aa40-5325c2bf2d47) |
| mint-compression-dist-xl.sh | :x: [more...](#3a98c4be-40bc-4ee6-bc27-698e2eac0b5c) |
<a name="52653185-79e2-430c-909b-a98854ea28c8"></a>
### 8220-26783b2/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30900
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dceafd8a1bd3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-o1ib1q6krolfb6",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365043,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15C487BDD56F9833, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="3a98c4be-40bc-4ee6-bc27-698e2eac0b5c"></a>
### 8220-26783b2/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30088
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 087aee84996e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 2 minutes and 6 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 48 seconds
(8/14) Running minio-dotnet tests ... done in 36 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 5 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 36 seconds
(11/14) Running minio-js tests ... done in 56 seconds
(12/14) Running minio-py tests ... FAILED in 4 minutes and 28 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "duration": 292,
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1396, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 152, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2039, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1414, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "args": {
    "bucket_name": "minio-py-test-9210f3ee-70a2-45b1-92f5-1feba3aee1bc",
    "presigned_get_object_url": "http://72.28.97.52:30088/minio-py-test-9210f3ee-70a2-45b1-92f5-1feba3aee1bc/b787f04c-e25c-48d6-a89e-7726dcc00d39?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190915%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190915T061640Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&response-content-language=en_US&response-content-type=text%2Fplain&X-Amz-Signature=947b4c47731ebc83025d32d8803c0f74cc67a7b3a9c97e3723f8f93554463762",
    "object_name": "b787f04c-e25c-48d6-a89e-7726dcc00d39"
  },
  "message": "'content-language'",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
<a name="c67f6841-037c-4069-aa40-5325c2bf2d47"></a>
### 8220-26783b2/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31270
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0bd17b572cd9:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5rihanrbf2sevk",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365923,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15C487BF075AA27B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,11 @@ checks:
 getdeps:
 	@mkdir -p ${GOPATH}/bin
 	@which golint 1>/dev/null || (echo "Installing golint" && GO111MODULE=off go get -u golang.org/x/lint/golint)
+ifeq ($(GOARCH),s390x)
+	@which staticcheck 1>/dev/null || (echo "Installing staticcheck" && GO111MODULE=off go get honnef.co/go/tools/cmd/staticcheck)
+else
 	@which staticcheck 1>/dev/null || (echo "Installing staticcheck" && wget --quiet https://github.com/dominikh/go-tools/releases/download/2019.2.3/staticcheck_${GOOS}_${GOARCH}.tar.gz && tar xf staticcheck_${GOOS}_${GOARCH}.tar.gz && mv staticcheck/staticcheck ${GOPATH}/bin/staticcheck && chmod +x ${GOPATH}/bin/staticcheck && rm -f staticcheck_${GOOS}_${GOARCH}.tar.gz && rm -rf staticcheck)
+endif
 	@which misspell 1>/dev/null || (echo "Installing misspell" && GO111MODULE=off go get -u github.com/client9/misspell/cmd/misspell)
 
 crosscompile:

